# State of Data Brazil

A State of Data Brazil foi realizada entre 18 de outubro de 2021 e 6 de dezembro de 2021 através de um questionário online e reuniu indicadores relacionados a perfil demográfico, formação, atuação no setor, remuneração, rotatividade e fatores de satisfação no ambiente de trabalho, incluindo o impacto do trabalho remoto nas preferências profissionais de 2.645 respondentes de todo o Brasil. A amostra reflete a visão de variados papéis de atuação em empresas, como os de analista de dados, cientista de dados e engenheiro de dados, bem como diferentes perfis de experiência profissional, incluindo analistas júnior, pleno, sênior e gestores.

Relatório: https://www.stateofdata.com.br

# Importação e primeiras analises do dado

In [1]:
import pandas as pd
import numpy as np

In [2]:
data = pd.read_csv('State of Data.csv')
data

/home/anyelle/.pyenv/versions/3.8.12/envs/lewagon/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3185: DtypeWarning: Columns (47) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


('P0', 'id')  ('P1_a ', 'Idade')  \
0     qkx4q0ei90wcjxnqkx4q0j3xgf0zn13s                38.0   
1     zdl2n19yhgpnoaco6kkczdl2nwv9zwrt                39.0   
2     vsamqp2un3q7us84mgvsams5fulsmcoh                30.0   
3     v31ab41botodnsv31zgg4k34zzojy81l                38.0   
4     ubhu8ntvm4xc0sfkdubhu80e973eek0k                36.0   
...                                ...                 ...   
2640  07nvo8cdy377a46gon207nvjebwe65zg                40.0   
2641  01mqnby32u01pg3nuj6vipp4mc06s8pg                27.0   
2642  01fncs8qbhl9zx01fncs8rqib2zl3d18                24.0   
2643  075h3v2z19if04yn7o075h3vhifl38rw                38.0   
2644  037usz1h2yip96dgil4mhj037uszw1gk                27.0   

     ('P1_a_a ', 'Faixa idade') ('P1_b ', 'Genero')  \
0                         35-39           Masculino   
1                         35-39           Masculino   
2                         30-34           Masculino   
3                         35-39            Feminino   
4                         35-39           Masculino   
...                         ...                 ...   
2640                      40-44           Masculino   
2641                      25-29           Masculino   
2642                      22-24           Masculino   
2643                      35-39           Masculino   
2644                      25-29            Feminino   

     ('P1_e ', 'Estado onde mora') ('P1_e_a ', 'uf onde mora')  \
0                       Ceará (CE)                          CE   
1                       Bahia (BA)                          BA   
2              Santa Catarina (SC)                          SC   
3                   São Paulo (SP)                          SP   
4              Santa Catarina (SC)                          SC   
...                            ...                         ...   
2640                São Paulo (SP)                          SP   
2641           Santa Catarina (SC)                          SC   
2642           Santa Catarina (SC)                          SC   
2643           Rio de Janeiro (RJ)                          RJ   
2644             Minas Gerais (MG)                          MG   

     ('P1_e_b ', 'Regiao onde mora') ('P1_g_b ', 'Regiao de origem')  \
0                           Nordeste                             NaN   
1                           Nordeste                         Sudeste   
2                                Sul                             NaN   
3                            Sudeste                             NaN   
4                                Sul                             NaN   
...                              ...                             ...   
2640                         Sudeste                         Sudeste   
2641                             Sul                             Sul   
2642                             Sul                             NaN   
2643                         Sudeste                             NaN   
2644                         Sudeste                             NaN   

      ('P1_g_c ', 'Mudou de Estado?') ('P1_h ', 'Nivel de Ensino')  ...  \
0                                   0                Pós-graduação  ...   
1                                   1                Pós-graduação  ...   
2                                   0                Pós-graduação  ...   
3                                   0                Pós-graduação  ...   
4                                   0                Pós-graduação  ...   
...                               ...                          ...  ...   
2640                                1   Não tenho graduação formal  ...   
2641                                1        Graduação/Bacharelado  ...   
2642                                0        Graduação/Bacharelado  ...   
2643                                0        Graduação/Bacharelado  ...   
2644                                0                Pós-graduação  ...   

     ('P8_d_g ', 'Criando e dando manutenção em ETLs, DAGs e 

In [3]:
data.columns

Index(['('P0', 'id')', '('P1_a ', 'Idade')', '('P1_a_a ', 'Faixa idade')',
       '('P1_b ', 'Genero')', '('P1_e ', 'Estado onde mora')',
       '('P1_e_a ', 'uf onde mora')', '('P1_e_b ', 'Regiao onde mora')',
       '('P1_g_b ', 'Regiao de origem')', '('P1_g_c ', 'Mudou de Estado?')',
       '('P1_h ', 'Nivel de Ensino')',
       ...
       '('P8_d_g ', 'Criando e dando manutenção em ETLs, DAGs e automações de pipelines de dados.')',
       '('P8_d_h ', 'Criando e gerenciando soluções de Feature Store e cultura de MLOps.')',
       '('P8_d_i ', 'Criando e mantendo a infra que meus modelos e soluções rodam (clusters, servidores, API, containers, etc.)')',
       '('P9_a ', 'Quais das iniciativas do Data Hackers que você já acessou/acompanhou?')',
       '('P9_a_a ', 'Blog/Medium do Data Hackers')',
       '('P9_a_b ', 'Podcast do Data Hackers')',
       '('P9_a_d ', 'Newsletter Semanal')', '('P9_a_e ', 'Canal do Slack')',
       '('P9_a_f ', 'Canal do Youtube do Data Hackers')',
     

In [4]:
data.shape

(2645, 356)

Como temos muitas colunas - 356, não conseguimos imprimir todas elas de forma simples. Para isso fazemos: 

In [5]:
# Analisando todas as colunas

with pd.option_context('display.max_rows', 2, 'display.max_columns', None): 
    display(data)

('P0', 'id')  ('P1_a ', 'Idade')  \
0     qkx4q0ei90wcjxnqkx4q0j3xgf0zn13s                38.0   
...                                ...                 ...   
2644  037usz1h2yip96dgil4mhj037uszw1gk                27.0   

     ('P1_a_a ', 'Faixa idade') ('P1_b ', 'Genero')  \
0                         35-39           Masculino   
...                         ...                 ...   
2644                      25-29            Feminino   

     ('P1_e ', 'Estado onde mora') ('P1_e_a ', 'uf onde mora')  \
0                       Ceará (CE)                          CE   
...                            ...                         ...   
2644             Minas Gerais (MG)                          MG   

     ('P1_e_b ', 'Regiao onde mora') ('P1_g_b ', 'Regiao de origem')  \
0                           Nordeste                             NaN   
...                              ...                             ...   
2644                         Sudeste                             NaN   

      ('P1_g_c ', 'Mudou de Estado?') ('P1_h ', 'Nivel de Ensino')  \
0                                   0                Pós-graduação   
...                               ...                          ...   
2644                                0                Pós-graduação   

     ('P1_i ', 'Área de Formação')  \
0                 Química / Física   
...                            ...   
2644                        Outras   

     ('P2_a ', 'Qual sua situação atual de trabalho?') ('P2_b ', 'Setor')  \
0                                      Empregado (CLT)          Marketing   
...                                                ...                ...   
2644                Desempregado, buscando recolocação                NaN   

     ('P2_c ', 'Numero de Funcionarios')  ('P2_d ', 'Gestor?')  \
0                           de 101 a 500                   1.0   
...                                  ...                   ...   
2644                                 NaN                   NaN   

     ('P2_e ', 'Cargo como Gestor') ('P2_f ', 'Cargo Atual')  \
0            Supervisor/Coordenador                      NaN   
...                             ...                      ...   
2644                            NaN                      NaN   

     ('P2_g ', 'Nivel')     ('P2_h ', 'Faixa salarial')  \
0                   NaN  de R$ 4.001/mês a R$ 6.000/mês   
...                 ...                             ...   
2644                NaN                             NaN   

     ('P2_i ', 'Quanto tempo de experiência na área de dados você tem?')  \
0                                       Mais de 10 anos                    
...                                                 ...                    
2644                                                NaN                    

     ('P2_j ', 'Quanto tempo de experiência na área de TI/Engenharia de Software você teve antes de começar a trabalhar na área de dados?')  \
0     Não tive experiência na área de TI/Engenharia ...                                                                                       
...                                                 ...                                                                                       
2644                                                NaN                                                                                       

      ('P2_k ', 'Você está satisfeito na sua empresa atual?')  \
0                                                   1.0         
...                                                 ...         
2644                                                NaN         

     ('P2_l ', 'Qual o principal motivo da sua insatisfação com a empresa atual?')  \
0                                                   NaN                              
...                                                 ...                              
2644                                                NaN                              

      ('P2_l_a '

In [6]:
colunas = data.columns.values.tolist()
colunas

["('P0', 'id')",
 "('P1_a ', 'Idade')",
 "('P1_a_a ', 'Faixa idade')",
 "('P1_b ', 'Genero')",
 "('P1_e ', 'Estado onde mora')",
 "('P1_e_a ', 'uf onde mora')",
 "('P1_e_b ', 'Regiao onde mora')",
 "('P1_g_b ', 'Regiao de origem')",
 "('P1_g_c ', 'Mudou de Estado?')",
 "('P1_h ', 'Nivel de Ensino')",
 "('P1_i ', 'Área de Formação')",
 "('P2_a ', 'Qual sua situação atual de trabalho?')",
 "('P2_b ', 'Setor')",
 "('P2_c ', 'Numero de Funcionarios')",
 "('P2_d ', 'Gestor?')",
 "('P2_e ', 'Cargo como Gestor')",
 "('P2_f ', 'Cargo Atual')",
 "('P2_g ', 'Nivel')",
 "('P2_h ', 'Faixa salarial')",
 "('P2_i ', 'Quanto tempo de experiência na área de dados você tem?')",
 "('P2_j ', 'Quanto tempo de experiência na área de TI/Engenharia de Software você teve antes de começar a trabalhar na área de dados?')",
 "('P2_k ', 'Você está satisfeito na sua empresa atual?')",
 "('P2_l ', 'Qual o principal motivo da sua insatisfação com a empresa atual?')",
 "('P2_l_a ', 'Falta de oportunidade de cresciment

## Organizando e identificando melhor as colunas

In [7]:
names_columns = []

for i in range(0, len(colunas)):
    names_columns.append(eval(colunas[i]))
    
names_columns

[('P0', 'id'),
 ('P1_a ', 'Idade'),
 ('P1_a_a ', 'Faixa idade'),
 ('P1_b ', 'Genero'),
 ('P1_e ', 'Estado onde mora'),
 ('P1_e_a ', 'uf onde mora'),
 ('P1_e_b ', 'Regiao onde mora'),
 ('P1_g_b ', 'Regiao de origem'),
 ('P1_g_c ', 'Mudou de Estado?'),
 ('P1_h ', 'Nivel de Ensino'),
 ('P1_i ', 'Área de Formação'),
 ('P2_a ', 'Qual sua situação atual de trabalho?'),
 ('P2_b ', 'Setor'),
 ('P2_c ', 'Numero de Funcionarios'),
 ('P2_d ', 'Gestor?'),
 ('P2_e ', 'Cargo como Gestor'),
 ('P2_f ', 'Cargo Atual'),
 ('P2_g ', 'Nivel'),
 ('P2_h ', 'Faixa salarial'),
 ('P2_i ', 'Quanto tempo de experiência na área de dados você tem?'),
 ('P2_j ',
  'Quanto tempo de experiência na área de TI/Engenharia de Software você teve antes de começar a trabalhar na área de dados?'),
 ('P2_k ', 'Você está satisfeito na sua empresa atual?'),
 ('P2_l ', 'Qual o principal motivo da sua insatisfação com a empresa atual?'),
 ('P2_l_a ', 'Falta de oportunidade de crescimento no emprego atual'),
 ('P2_l_b ', 'Salário a

In [8]:
df = pd.DataFrame(names_columns)
df.rename(columns = {0:'Categorias', 1:'Perguntas'}, inplace = True)

In [9]:
df

Categorias                          Perguntas
0           P0                                 id
1        P1_a                               Idade
2      P1_a_a                         Faixa idade
3        P1_b                              Genero
4        P1_e                    Estado onde mora
..         ...                                ...
351    P9_a_b             Podcast do Data Hackers
352    P9_a_d                  Newsletter Semanal
353    P9_a_e                      Canal do Slack
354    P9_a_f    Canal do Youtube do Data Hackers
355    P9_a_g   Ainda não conhecia o Data Hackers

[356 rows x 2 columns]

**Podemos perceber que o dado é dividido em categorias, de P0 a P9. Cada categoria é relacionada com um tipo de pergunta diferente.**

- P0: Id de cada usuário que respondeu ao questionário.
- P1: Perguntas pessoais e demográficas: Idade, Estado, Genero, ... **df_1**
- P2: Perguntas relacionadas ao emprego atual: Salario, Cargo, Nível, ... **df_2**
- P3: Perguntas mais específicas sobre o emprego atual: Nível da equipe, ... **df_3**
- P4: Perguntas sobre como é a analise de dados na empresa, que tipo de dado é utilizado. **df_4**
- P5: Perguntas sobre novas oportunidades (Apenas 3 perguntas (?)) **df_5**
- P6: Perguntas sobre rotina de trabalho relacionada a Engenharia de Dados e ferramentas utilizadas **df_6**
- P7: Perguntas sobre rotina de trabalho relacionada a Análise de Dados e ferramentas utilizadas **df_7**
- P8: Perguntas sobre rotina de trabalho relacionada a Ciencia de Dados e ferramentas utilizadas **df_8**
- P9: Perguntas sobre interação com Data Hackers **df_9**

Há a necessidade de dividir os dados em diferentes dataframes?

In [10]:
df_1 = data.loc[:,"('P1_a ', 'Idade')":"('P1_i ', 'Área de Formação')"]
df_1.insert(loc=0, column="('P0', 'id')", value=data["('P0', 'id')"])

In [11]:
df_2 = data.loc[:,"('P2_a ', 'Qual sua situação atual de trabalho?')":"('P2_s ', 'Caso sua empresa decida pelo modelo 100% presencial qual será sua atitude?')"]
df_2.insert(loc=0, column="('P0', 'id')", value=data["('P0', 'id')"])

In [12]:
df_3 = data.loc[:,"('P3_a ', 'Qual o número aproximado de pessoas que atuam com dados na sua empresa hoje?')":"('P3_d_n ', 'Dividir o tempo entre entregas técnicas e gestão.')"]
df_3.insert(loc=0, column="('P0', 'id')", value=data["('P0', 'id')"])

In [13]:
df_4 = data.loc[:,"('P4_a_a ', 'Mesmo que esse não seja seu cargo formal, você considera que sua atuação no dia a dia, reflete alguma das opções listadas abaixo?')":"('P4_h_x ', 'Não utilizo nenhuma ferramenta de BI no trabalho')"]
df_4.insert(loc=0, column="('P0', 'id')", value=data["('P0', 'id')"])

In [14]:
df_5 = data.loc[:,"('P5_b ', 'Qual oportunidade você está buscando?')":"('P5_d ', 'Como tem sido a busca por um emprego na área de dados?')"]
df_5.insert(loc=0, column="('P0', 'id')", value=data["('P0', 'id')"])

In [15]:
df_6 = data.loc[:,"('P6_a ', 'Quais das opções abaixo fazem parte da sua rotina no trabalho atual como engenheiro de dados?')":"('P6_h_h ', 'Cuidando da qualidade dos dados, metadados e dicionário de dados.')"]
df_6.insert(loc=0, column="('P0', 'id')", value=data["('P0', 'id')"])

In [16]:
df_7 = data.loc[:,"('P7_a ', 'Quais das opções abaixo fazem parte da sua rotina no trabalho atual com análise de dados?')":"('P7_d_h ', 'Utilizando ferramentas avançadas de estatística como SAS, SPSS, Stata etc, para realizar análises.')"]
df_7.insert(loc=0, column="('P0', 'id')", value=data["('P0', 'id')"])

In [17]:
df_8 = data.loc[:,"('P8_a ', 'Quais das opções abaixo fazem parte da sua rotina no trabalho atual com ciência de dados?')":"('P8_d_i ', 'Criando e mantendo a infra que meus modelos e soluções rodam (clusters, servidores, API, containers, etc.)')"]
df_8.insert(loc=0, column="('P0', 'id')", value=data["('P0', 'id')"])

In [18]:
df_9 = data.loc[:,"('P9_a ', 'Quais das iniciativas do Data Hackers que você já acessou/acompanhou?')":"('P9_a_g ', 'Ainda não conhecia o Data Hackers')"]
df_9.insert(loc=0, column="('P0', 'id')", value=data["('P0', 'id')"])